maven-release-manager/src/main/java/org/apache/maven/shared/release/phase/AbstractRewritePomsPhase.java

```diff
@@ -96,7 +96,7 @@ public abstract class AbstractRewritePomsPhase extends AbstractReleasePhase impl
      * Regular expression pattern matching Maven expressions (i.e. references to Maven properties).
      * The first group selects the property name the expression refers to.
      */
-    private static final Pattern EXPRESSION_PATTERN = Pattern.compile("\\$\\{(.+)\\}");
+    private static final Pattern EXPRESSION_PATTERN = Pattern.compile("\\$\\{(.+?)\\}");
```

Comment:
Is .+? the same as .*

Reply:
No it isn’t: https://docs.oracle.com/javase/tutorial/essential/regex/quant.html. The former is greedy while the latter is reluctant.